pkg/controller/ttlafterfinished/ttlafterfinished_controller.go

```diff
@@ -235,30 +239,34 @@ func (tc *Controller) processJob(key string) error {
 		Preconditions:     &metav1.Preconditions{UID: &fresh.UID},
 	}
 	klog.V(4).Infof("Cleaning up Job %s/%s", namespace, name)
-	return tc.client.BatchV1().Jobs(fresh.Namespace).Delete(context.TODO(), fresh.Name, options)
+	if err := tc.client.BatchV1().Jobs(fresh.Namespace).Delete(context.TODO(), fresh.Name, options); err != nil {
+		return err
+	}
+	metrics.TimeToDeletionSeconds.Observe(time.Since(*expiredAt).Seconds())
+	return nil
 }
 
 // processTTL checks whether a given Job's TTL has expired, and add it to the queue after the TTL is expected to expire
 // if the TTL will expire later.
-func (tc *Controller) processTTL(job *batch.Job) (expired bool, err error) {
+func (tc *Controller) processTTL(job *batch.Job) (expired bool, expiredAt *time.Time, err error) {
```

Comment:
That seems like an overkill, you can easily express if job expired through `expiredAt` being set, when it's `nil` it did not. I don't see a reason to have too many return variables.

Reply:
done.